### PR TITLE
LPS-125089 - Content is temporarily unavailable in Date field after editing in Spreadsheet View

### PR DESCRIPTION
**Pull request by victorcdp**
Relevant Issue: [LPS-125089](https://issues.liferay.com/browse/LPS-125089)
The issue is occurring because the date is being saved with a bunch of extra characters ( ' " ', '[', ']') when updated through the spreadsheet view.
Proposed fix is to filter the string before saving.

### CODE REVIEW

**Comment by victorcdp** [comment]
ci:test:sf
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/java/com/liferay/dynamic/data/lists/web/internal/portlet/action/UpdateRecordMVCResourceCommand.java [hunk 1]
```diff
@@ -124,9 +125,22 @@ private Map<String, List<DDMFormFieldValue>> _toDDMFormFieldValues(
...
+					if (jsonObject.getString(
+							"name"
+						).equals(
+							"eventDate"
+						)) {
+
+						valueJSONObjectString = StringUtil.removeChars(
+							valueJSONObjectString, '[', ']', '"');
+					}
+
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/java/com/liferay/dynamic/data/lists/web/internal/portlet/action/UpdateRecordMVCResourceCommand.java:140]: @victorcdp Unfortunately this solution can cause other problems. Imagine if the user adds a New Data Definition with a text field named  `eventDate`..

image

With this solution we will also remove these characters from the text field, when we should just remove them from the date field.

Comment (carolmariaabb) [reply modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/java/com/liferay/dynamic/data/lists/web/internal/portlet/action/UpdateRecordMVCResourceCommand.java:140]: Can you check if we can fix the bug on the frontend side? You can start investigating this file [here](https://github.com/liferay/liferay-portal/blob/db7aab4b87b4f63f3bcdc90d409c220791e1dae8/modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/resources/META-INF/resources/js/main.js#L642-L668).

Thanks!

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by carolmariaabb** [comment]
@victorcdp Please see my comments below...I'll close this PR for now, but feel free to reach out to the team if you need any help.

**Comment by carolmariaabb** [comment]
ci:close

**Pull request closed without merging**